### ISSUE TITLE
No `invalid-type-form` errors for invalid type forms in quoted annotations

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#2553]
### Summary

The typing conformance suite points out that we are supposed to emit an error here complaining about an invalid type form, but we do not; instead, we just silently infer `Unknown` for the `x` variable:

```py
x: "1234"
```

We do emit an error for the same expression if it is not quoted:

```py
x: 1234  # error [invalid-type-form] "Int literals are not allowed in this context in a type expression"
```
